niceman/support/distributions/debian.py

```diff
@@ -41,27 +38,36 @@ class DebianReleaseSpec(object):
     codename = attr.ib()
     suite = attr.ib()
     version = attr.ib()
-    date = attr.ib()     # in XXX format?
-    components = attr.ib()#validator=attr.validators.instance_of(list),convert=str.split),
-    architectures = attr.ib()#validator=attr.validators.instance_of(list),convert=str.split)
+    date = attr.ib()
+    components = attr.ib()
+    architectures = attr.ib()
 
 
-def get_spec_from_release_file(release_filename):
+def get_spec_from_release_file(content):
     """Provide specification object describing the component of the distribution
-
-    Examples
-    --------
-
-     SKIP TST FOR NOW >>> get_spec_from_release_file('/var/lib/apt/lists/neuro.debian.net_debian_dists_jessie_InRelease')
-
     """
-    release = deb822.Release(codecs.open(release_filename, 'r', 'utf-8'))
+    # RegExp to pull a single line "tag: value" pair from a deb822 file
+    re_deb822_single_line_tag = re.compile("""
+        ^(?P<tag>[a-zA-Z][^:]*):[\ ]+  # Tag - begins at start of line
+        (?P<val>\S.*)$           # Value - after colon to the end of the line
+    """, flags=re.VERBOSE + re.MULTILINE)
+
+    # Split before PGP signature if present
+    if "-----BEGIN PGP SIGNATURE-----" in content:
+        content = content.split("-----BEGIN PGP SIGNATURE-----")[0]
```

Comment:
couldn't it be without 'if' statement and just the `content = content.split("-----BEGIN PGP SIGNATURE-----")[0]`? ;-)

Reply:
Look at that!  That's nice behavior for str.split()